### ISSUE TITLE
Data validation on Nodes entry page

### ISSUE THREAD

**Issue by jsayers-bsc** [MotusDev/Motus-TO-DO#660]
I recently entered some CTT Nodes into a deployment record for the first time and noticed that an invalid entry was permitted. Then since the page is pretty new I just tried a variety of entries to see if they would be permitted.

- Invalid Lat Lon values (outside the range of true options) are currently permitted. 
- Blank or missing Lat Lon fields are also permitted
- Text in the Lat Lon fields is permitted, but the field reverts to nulls after saved
- Multiple Nodes at identical Lat Lons are permitted. I'm not sure this is an issue, or how close to another node a node would be installed, but identical Lat Lons probably wouldn't happen in real life, and may indicate data entered in error.
- Invalid entries in the ID field (such as punctuation or nulls) are permitted

**Comment by denislepage** [comment]
Might be useful to throw an error or warning I'd coordinates for a node is too far from the base ( and/or show the distance)

I might prefer preventing obviously bad entries than relying solely on the notification center.

**Comment by denislepage** [comment]
I also think we should allow missing coordinates for situation where node may be moved regularly. I would rather know that nodes are present, even if we can't support the entry of moving coordinates.

Newer nodes should have built in gps capabilities, anyway.

**Comment by pmorrill** [comment]
This entry form was put together very quickly, without any real requirements. It definitely needs error checks and some stream-lining

**Comment by jsayers-bsc** [comment]
> 
> 
> This entry form was put together very quickly, without any real requirements. It definitely needs error checks and some stream-lining

Yes that was definitely clear! Hopefully my various attempts to mess with the data entry have helped cover the bases for validation.

**Comment by jsayers-bsc** [comment]
After some testing...

- Invalid ID (containing characters other than letters or numbers) throws an error and does not save the record. 

- Invalid lat lon (out of bounds) displays an error for each, but _does_ save the record (though with null lat lon values). 

- Invalid lat lon (non numbers in lat lon fields) saves without an error message (but with blank lat lon). I think we'd want an invalid location error here?

- Missing lat lon throws no error (as expected since we discussed earlier that we should permit for now)

- Identical lat lon as another node is permitted. Is that an issue? Not sure.

As an aside, I noticed in the email from David containing the node data, that letters are all in upper case.
